src/Symfony/Component/Console/Tests/TerminalTest.php

```diff
@@ -93,4 +93,47 @@ public function testSttyOnWindows()
         $terminal = new Terminal();
         $this->assertSame((int) $matches[1], $terminal->getWidth());
     }
+
+    public function testGetTermColorSupport()
+    {
+        $colorterm = getenv('COLORTERM');
+        $term = getenv('TERM');
+
+        try {
+            putenv('COLORTERM=truecolor');
+            putenv('TERM');
+            $this->assertSame('Ansi24', Terminal::getTermColorSupport());
+
+            putenv('COLORTERM=TRUECOLOR');
+            putenv('TERM');
+            $this->assertSame('Ansi24', Terminal::getTermColorSupport());
+
+            putenv('COLORTERM=somethingLike256Color');
+            putenv('TERM');
+            $this->assertSame('Ansi8', Terminal::getTermColorSupport());
```

Comment:
Those should be separate tests (or maybe using a data provider to implement them, passing values for `COLORTERM` and `TERM` and the expected support)

Reply:
Done